simulator/src/main/java/com/hazelcast/simulator/coordinator/deployment/DeploymentPlan.java

```diff
@@ -22,75 +22,57 @@
 import com.hazelcast.simulator.protocol.registry.AgentData;
 import com.hazelcast.simulator.protocol.registry.ComponentRegistry;
 import com.hazelcast.simulator.worker.WorkerType;
-import org.apache.log4j.Logger;
 
-import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
 
-import static com.hazelcast.simulator.coordinator.deployment.DeploymentUtils.formatIpAddresses;
 import static com.hazelcast.simulator.coordinator.deployment.DeploymentUtils.generateFromArguments;
 import static com.hazelcast.simulator.coordinator.deployment.DeploymentUtils.generateFromXml;
-import static com.hazelcast.simulator.utils.FormatUtils.HORIZONTAL_RULER;
-import static com.hazelcast.simulator.utils.FormatUtils.formatLong;
-import static com.hazelcast.simulator.utils.FormatUtils.padLeft;
-import static java.lang.String.format;
 
 public final class DeploymentPlan {
 
-    private static final int WORKER_MODE_LENGTH = 6;
-
-    private static final Logger LOGGER = Logger.getLogger(DeploymentPlan.class);
-
     private final Set<String> versionSpecs = new HashSet<String>();
 
-    private final List<AgentWorkerLayout> agentWorkerLayouts;
+    private final Map<SimulatorAddress, List<WorkerProcessSettings>> workerDeployment;
     private final int memberWorkerCount;
     private final int clientWorkerCount;
 
-    private DeploymentPlan(List<AgentWorkerLayout> agentWorkerLayouts) {
+    private DeploymentPlan(Map<SimulatorAddress, List<WorkerProcessSettings>> workerDeployment) {
```

Comment:
It's a private constructor, so an implementation detail. But it's the list of workers to be deployed, the "diff" between the old and new state. It's directly in the return format of your old `asMap()` method.